### PR TITLE
fix(rtl): flip arrow icons on rtl

### PR DESCRIPTION
**Pull request by AmitMY**
#### Short description of what this resolves:
bad arrow direction on rtl

#### Changes proposed in this pull request:

- flip arrow icons on rtl

#### Only arrows?
I understand that more icons should be flipped, for example, `exit`, which is also an arrow, but that should be somehow in the icon's metadata.
I went all out in [this edit](https://github.com/AmitMY/ionic/blob/b8e690af8a108f7e81085b3e8fcaad8c3fc90920/src/fonts/ionicons.scss) and flipped all icons that can be flipped without causing harm, so this is just a toned down version

**Ionic Version**: 3.x

### CODE REVIEW

**Comment by brandyscarney** [comment]
@AmitMY What if we had an attribute that only flips the icon for `rtl` if it is added. Something like this:

```
[dir="rtl"] [flip-rtl] {
  -moz-transform: scaleX(-1);
  -o-transform: scaleX(-1);
  -webkit-transform: scaleX(-1);
  transform: scaleX(-1);
}
``` 

Then if you want the icon to be flipped you would add it like:

```
<ion-icon name="arrow-forward" flip-rtl></ion-icon>
```

This would make it completely up to the user to flip the icon or not, could be used on icons outside of ionicons, and gets rid of any hardcoded styles for ionicons in case icons are added or removed. Then we would need to add some styling for our icons that are used in components. What do you think of this solution? Would it be viable?

**Comment by AmitMY** [comment]
@brandyscarney While I think that should be an option (the `[flip-rtl]` toggles flipping), I also think that some icons, and at least the arrow ones should be auto flipped.

It is a hassle, so most users want use it.

The arrows mostly use `forward` and `back`, and that agrees with the style of `start` and `end`. No implicit left and right, so it needs to be flipped on rtl...

Look for the word "arrow" here:
https://ionicframework.com/docs/ionicons/

File: src/fonts/ionicons.scss [hunk 1]
```diff
@@ -33,3 +33,20 @@ ion-icon {
   text-transform: none;
   speak: none;
 }
+
+@mixin scale-x($amount) {
```
Comment (manucorporat) [inline src/fonts/ionicons.scss]: scss already does it for transform, `scale-x` is not needed

Comment (AmitMY) [inline src/fonts/ionicons.scss]: @manucorporat can you please explain a bit more? If I only do `transform: scaleX(-1);` it does not cover browsers support, so I am not sure what you mean.

Comment (AmitMY) [inline src/fonts/ionicons.scss]: @manucorporat I looked in some code, and it seems you never do custom browser support for `transform`, so I guess I'll do the same. Updated

**Comment by AmitMY** [comment]
@manucorporat Anything else to fix here?

**Comment by AmitMY** [comment]
@brandyscarney Can you re-review?

**Pull request closed without merging**